coordinator/src/db/channels.rs

```diff
@@ -77,6 +80,26 @@ pub(crate) fn get_all_non_pending_channels(conn: &mut PgConnection) -> QueryResu
         .load(conn)
 }
 
+pub(crate) fn update_payment_hash(
+    payment_hash: PaymentHash,
+    funding_txid: String,
+    conn: &mut PgConnection,
+) -> Result<()> {
+    let channel = channels::table
+        .filter(channels::funding_txid.eq(funding_txid.clone()))
+        .first(conn)
+        .optional();
```

Comment:
🙃 you could skip that `optional()` and simplify the match below.

Without optional you would get a `NotFound` Error instead of `None`.